### DISCUSSION TITLE
Dependency injection not working with a singleton inside of a command file?

### DISCUSSION THREAD

**Discussion by arvcork** [adonisjs/core#4730] [Help]
Hi all,

I am reaching out to get some support with the IoC and how it works with commands.

I have a singleton, which is registered with the container. I can confirm this by adding some logs in there and can see them in the terminal once it has been started. 

The idea is, I have a singleton which is for all Discord interaction from my API, to avoid having to do multiple login requests etc and when resolving this from the IoC it'll be the instance which has the clients, guilds etc and I can interact with them from there accordingly.

However, when trying to use this within a command, which is used to sync roles it just doesn't seem to be working.

I am binding it like so:

providers/discord_provider.ts
```
export default class DiscordProvider {
  constructor(protected app: ApplicationService) {}

  /**
   * Register bindings to the container
   */
  register() {
    this.app.container.bind('App/Discord', () => {
      const apiKey = this.app.config.get<string>('app.discordApiKey')
      return new DiscordBot(apiKey)
    })
  }
}
```

Then inside of the command, I am just doing something as simple as the following:

```
async run() {
  const bot = await this.app.container.make('App/Discord')
  const tempServer = bot.findGuild('xYz')

  if (tempServer === undefined) {
    this.logger.error('Unable to sync due to server not being able to be found')
    return
  }

  this.logger.info(`Syncing ${tempServer.name}`)
}
```

I have tried this the manual way but have also tried to do the @inject decorator as well, as per the docs but haven't managed to have any successful attempt.

The error which I get when doing it manually is:

> RuntimeException: Cannot resolve binding "App/Discord" from the container

If I do it using the decorator I get the following:

> RuntimeException: Cannot inject "[class DiscordBot]" in "[class DiscordSyncRoles]"
> Container is not able to resolve "DiscordSyncRoles" class dependencies. Did you mean to use @inject decorator

Apologies if this is quite a nooby question, I've tried Stack Overflow and also looking at the documentation a lot but can't seem to come to any resolution, so any help is much appreciated! 🙏

**Comment by arvcork** [comment]
It's probably worth mentioning as well, I have also tried:

```
this.app.container.singleton('App/Discord', ...)
```

**Comment by thetutlage** [comment]
Seems like you are using v5. If so, please load the app when running the command using the following option. https://v5-docs.adonisjs.com/guides/ace-commandline#booting-the-app-within-the-command

```ts
public static settings = {
  loadApp: true
}
```

**Reply by arvcork** [reply]
I'm using v6 as far as I am aware, in package.json `"@adonisjs/core": "^6.12.1"`.

I actually came across the `startApp` option and set that to `true` but unfortunately, I still get the same error.

**Reply by arvcork** [reply]
Ahh, okay... 

Manual binding seems to work this way, but not the @inject decorator.
